From a12bec9a73c4b74a7f96a8d1a67c1c42f9040faa Mon Sep 17 00:00:00 2001
From: Kiran Kumar K <kirankumark@marvell.com>
Date: Mon, 6 Nov 2023 14:56:45 +0530
Subject: [PATCH 757/955] common/cnxk: fix issue with flow key alg for cnxk

Fix issue with flow key alg generation for cnxk device
in case of represented port.

Fixes: ed43baf1a ("drivers: add represented port flow item for cnxk")

Signed-off-by: Kiran Kumar K <kirankumark@marvell.com>
Change-Id: Ifa68e5d585ebc645fbc1d6e29df284306aac8171
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/dataplane/dpdk/+/115334
Base-Builds: sa_ip-toolkits-Jenkins <sa_ip-toolkits-jenkins@marvell.com>
Base-Tests: sa_ip-toolkits-Jenkins <sa_ip-toolkits-jenkins@marvell.com>
Tested-by: sa_ip-toolkits-Jenkins <sa_ip-toolkits-jenkins@marvell.com>
Reviewed-by: Jerin Jacob Kollanukkaran <jerinj@marvell.com>
(cherry picked from commit c934cdf8c3665f79ae4395bc88957b64128efb54)
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/dataplane/dpdk/+/119882
Reviewed-by: Satheesh Paul Antonysamy <psatheesh@marvell.com>
Tested-by: Jerin Jacob <jerinj@marvell.com>
---
 drivers/common/cnxk/roc_npc.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/common/cnxk/roc_npc.c b/drivers/common/cnxk/roc_npc.c
index 7076c4e2c3f0a..65ed39a266625 100644
--- a/drivers/common/cnxk/roc_npc.c
+++ b/drivers/common/cnxk/roc_npc.c
@@ -1113,8 +1113,10 @@ npc_rss_action_program(struct roc_npc *roc_npc,
 	uint8_t alg_idx;
 	int rc;
 
-	if (flow->has_rep)
+	if (flow->has_rep) {
 		npc = roc_npc->rep_npc;
+		npc->flowkey_cfg_state = roc_npc->flowkey_cfg_state;
+	}
 
 	for (; actions->type != ROC_NPC_ACTION_TYPE_END; actions++) {
 		if (actions->type == ROC_NPC_ACTION_TYPE_RSS) {
-- 
2.25.1

